### PR TITLE
Feature: Added TypeScript Support for Directus Instance

### PR DESCRIPTION
**Pull request by Fritsch-Tech**
#### 🔧 Types of changes

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update
- [ ] Codebase improvement
- [ ] Other (if none of the other choices apply)

#### 🚨 Checklist

- [x] I've read the [guidelines for contributing](https://github.com/gremo/react-directus/blob/main/CONTRIBUTING.md)
- [x] I've added necessary documentation (if appropriate)
- [x] I've ensured that my code additions do not fail linting or unit tests (if applicable)

#### Description

Description

This pull request introduces TypeScript support for the Directus instance, as described in the [Directus SDK documentation](https://docs.directus.io/reference/old-sdk.html#typescript). By adding TypeScript support, we can now benefit from enhanced type safety and improved code intelligence when interacting with the Directus instance in our codebase.

This enhancement allows developers to take full advantage of TypeScript's static type checking, catching potential errors early in the development process and providing better developer experience.

### CODE REVIEW

File: src/types.ts [hunk 1]
```diff
@@ -1,5 +1,5 @@
 import * as React from 'react';
-import { DirectusOptions, IDirectus } from '@directus/sdk';
+import { DirectusOptions, IDirectus, TypeMap } from '@directus/sdk';
```
Comment (gremo) [inline src/types.ts:2]: Is TypeMap available for SDK 9 and 10 right?

File: src/types.ts [hunk 2]
```diff
@@ -67,12 +67,14 @@ export interface DirectusProviderProps {
 /**
  * Shape of the main context.
  */
-export interface DirectusContextType {
+export interface DirectusContextType<T extends TypeMap> {
   apiUrl: string;
   // eslint-disable-next-line @typescript-eslint/no-explicit-any
-  directus: IDirectus<any>;
+  directus: IDirectus<T>;
```
Comment (gremo) [inline src/types.ts:73]: Can `T = any` by default, so legacy code doesn't to be changed? If we leave `IDirectus<T>` people is then forced to update the code, right?

File: src/types.ts (hunk 1 above)
Comment (Fritsch-Tech) [reply src/types.ts:2]: Yes, as documented in [DirectusSDK9 Types](https://www.npmjs.com/package/@directus/sdk/v/9.14.2?activeTab=code).

File: src/types.ts (hunk 2 above)
Comment (Fritsch-Tech) [reply src/types.ts:73]: I can set a `TypeMap` as the default type for the Directus provider, so legacy code won't be afected in any way.

**Comment by gremo** [comment]
Thanks, this is a nice addition!

**Pull request merged**